fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Election.java

```diff
@@ -180,26 +192,17 @@ public long getEndTimestamp() {
     return end;
   }
 
-  /**
-   * Computes the hash for the registered votes, when terminating an election (sorted by message
-   * id's alphabetical order)
-   *
-   * @return the hash of all registered votes
-   */
-  public String computerRegisteredVotes() {
-    List<String> listOfVoteIds = new ArrayList<>();
-    // Since messageMap is a TreeMap, votes will already be sorted in the alphabetical order of
-    // messageIds
-    for (PublicKey senderPk : messageMap.values()) {
-      for (ElectionVote vote : voteMap.get(senderPk)) {
-        listOfVoteIds.add(vote.getId());
-      }
+  public void putEncryptedVotesBySender(PublicKey senderPk, List<ElectionEncryptedVote> votes) {
+    if (senderPk == null) {
+      throw new IllegalArgumentException("Sender public key cannot be null.");
     }
-    if (listOfVoteIds.isEmpty()) {
-      return "";
-    } else {
-      return Hash.hash(listOfVoteIds.toArray(new String[0]));
+    if (votes == null || votes.isEmpty()) {
+      throw new IllegalArgumentException("Encrypted votes cannot be null or empty");
     }
+    // The list must be sorted by order of question ids
+    List<ElectionEncryptedVote> votesCopy = new ArrayList<>(votes);
+    votesCopy.sort(Comparator.comparing(ElectionEncryptedVote::getQuestionId));
+    encryptedVoteByPublicKey.put(senderPk, votesCopy);
```

Comment:
Sorry for the misunderstanding, it should be ordered by `vote_id`, not question id.

As described here https://github.com/dedis/popstellar/pull/1081/files#diff-b577fe0224ef66435ee10949840f99a9f91194dc18e0138c16fe15f5c8190500

or implemented here:

https://github.com/dedis/popstellar/pull/1081/files#diff-66a954b5689d1a6ede24f210ef79643d6fa6c904997e5d07634b6a58abe4997d